nextstrain/cli/runner/managed_conda.py

```diff
@@ -0,0 +1,337 @@
+    machine = platform.machine()
+
+    if (system, machine) == ("Linux", "x86_64"):
+        subdir = "linux-64"
+    elif (system, machine) in {("Darwin", "x86_64"), ("Darwin", "arm64")}:
+        subdir = "osx-64"
+    else:
+        warn(f"Unsupported system/machine: {system}/{machine}")
+        return False
+
+    # Releases have other attributes related to system/machine, but they're
+    # informational-only and subdir is what Conda *actually* uses to
+    # differentiate distributions/files/etc.  Use it too so we have the same
+    # view of reality.
+    dist = next((d for d in dists if d.get("attrs", {}).get("subdir") == subdir), None)
+
+    assert dist, f"unable to find micromamba dist with subdir == {subdir!r}"
+
+    # download_url is scheme-less, so add our preferred scheme but in a way
+    # that won't break if it starts including a scheme later.
+    dist_url = urljoin("https:", dist["download_url"])
+
+    print(f"Requesting Micromamba from {dist_url}…")
+
+    response = requests.get(dist_url, stream = True)
+    content_type = response.headers["Content-Type"]
+
+    assert content_type == "application/x-tar", \
+        f"unknown content-type for micromamba dist: {content_type}"
+
+    with tarfile.open(fileobj = response.raw, mode = "r|*") as tar:
+        # Ignore archive members starting with "/" and or including ".." parts,
+        # as these can be used (maliciously or accidentally) to overwrite
+        # unintended files (e.g. files outside of MICROMAMBA_ROOT).
+        safe_members = (
+            member
+                for member in tar
+                 if not member.name.startswith("/")
+                and ".." not in PurePosixPath(member.name).parts)
+
+        print(f"Downloading and extracting Micromamba to {MICROMAMBA_ROOT}…")
+        tar.extractall(path = str(MICROMAMBA_ROOT), members = safe_members)
+
+    return True
+
+
+def setup_prefix(force: bool = False) -> bool:
+    """
+    Install Conda packages with Micromamba into our ``PREFIX``.
+    """
+    if not force and (PREFIX_BIN / "augur").exists():
+        print(f"Using existing Conda packages in {PREFIX}.")
+        print(f"  Hint: if you want to ignore this existing installation, re-run `nextstrain setup` with --force.")
+        return True
+
+    if PREFIX.exists():
+        print(f"Removing existing directory {PREFIX} to start fresh…")
+        shutil.rmtree(str(PREFIX))
+
+    # Conda packages to install, based on our unmanaged "native" install docs.
+    #
+    # Omits nextstrain-cli as that's us and we're already installed and
+    # running. ;-)
+    #
+    # Adds bash so that a newer version is guaranteed to be available on
+    # systems with an older bash.  This is similar to how our Docker runtime
+    # image includes its own bash too.
+    #
+    # HEY YOU: If you add/remove packages here, make sure to account for how
+    # update() should make the same changes to existing envs.
+    #   -trs, 1 Sept 2022
+    packages = (
```

Comment:
Looking at these packages it seems set up for most basic pipelines + nCoV. What's the story for a pipeline that has dependencies beyond these? Self-installing them kind-of defeats the purpose of a managed runtime (and would require inside knowledge of install locations as, in my case, I'd need something like `~/Library/Application\ Support/nextstrain/cli/micromamba/bin/micromamba --prefix ~/Library/Application\ Support/nextstrain/cli/runtimes/managed-conda/ some_package`). I guess this is similar for docker. Is the thinking that this is good enough for now?

Reply:
> I guess this is similar for docker. Is the thinking that this is good enough for now?

Yes, good enough for now, and similar to how our Docker runtime works at the moment. We wouldn't be expecting users to mess with the managed Conda env at all.

There's a longer-term story for how to support arbitrary workflow deps for both the Docker and this managed Conda runtimes, but that's for later.